Makefile

```diff
@@ -397,6 +397,10 @@ ci-job-readme-check:
 ci-job-clippy:
 	$(call banner,CI-Job: Clippy)
 	@cargo clippy -- -D warnings
+	# Run `cargo clippy` in select boards so we run clippy with targets that
+	# actually check the arch-specific functions.
+	@cd boards/nordic/nrf52840dk && cargo clippy -- -D warnings
+	@cd boards/hifive1 && cargo clippy -- -D warnings
```

Comment:
I don't like this, I feel like this is destined to go out of sync. That being said, I don't have a better idea. This is, again, caused by conditional compilation, and so I don't have a better idea that does not involve adding a new `#[cfg(any(clippy, ...` -- for which I know there will be opposition (https://github.com/tock/tock/pull/4059#discussion_r1663407997).

Reply:
Per usual there is a fix but it's not stable https://doc.rust-lang.org/cargo/reference/unstable.html#per-package-target

Reply:
I tried out that flag and it seems to be everything I want. `cargo clippy` and `cargo check` just work as expected, and `cargo test` fails because we don't have any way to test on an embedded device.

Reply:
I presume it would totally break `hail` on stable, as it would depend on the `cortex-m` crates and those can't then be compiled with a stable Cargo?

Reply:
I think so.

Reply:
Yes.
```
❯ make
error: failed to load manifest for workspace member `/Users/bradjc/git/tock/arch/cortex-m`
referenced by workspace at `/Users/bradjc/git/tock/Cargo.toml`

Caused by:
  failed to parse manifest at `/Users/bradjc/git/tock/arch/cortex-m/Cargo.toml`

Caused by:
  the cargo feature `per-package-target` requires a nightly version of Cargo, but this is the `stable` channel
  See https://doc.rust-lang.org/book/appendix-07-nightly-rust.html for more information about Rust release channels.
  See https://doc.rust-lang.org/cargo/reference/unstable.html#per-package-target for more information about using this feature.
make: *** [/Users/bradjc/git/tock/target/thumbv7em-none-eabi/release/hail] Error 101
```.